main.go

```diff
@@ -166,21 +176,58 @@ func httpDo(req *http.Request) (*http.Response, error) {
 		time.Sleep(backOff * time.Duration(retryCount))
 	}
 
-	return nil, fmt.Errorf("giving up on %s", url)
+	return nil, fmt.Errorf("giving up on %s", req.URL)
 }
 
-func root(_ *cobra.Command, _ []string) {
-	if url == "" {
-		url = fmt.Sprintf("https://github.com/%s.keys", username)
+func httpWorker(urls chan *url.URL, results chan keyResponse) {
+	for url := range urls {
+		keyResult := keyResponse{URL: url}
+
+		// as james will exit in just a second - we accept context leakage
```

Comment:
I think what needs to be done is to wrap this up in a function of its own instead of all the `continue` statements - that way - defer could call cancel nicely